File: app/src/main/java/com/zhangw/aliencat/ui/welcome/WelcomeActivity.java
package com.zhangw.aliencat.ui.welcome;

import android.os.Bundle;
import android.widget.TextView;

import com.zhangw.aliencat.R;
import com.zhangw.aliencat.base.BaseActivity;

import java.util.concurrent.TimeUnit;

import butterknife.BindView;
import io.reactivex.Observable;
import io.reactivex.functions.Consumer;

/**
 * @author zhangw
 * @date 2018/3/23.
 */
public class WelcomeActivity extends BaseActivity {
    private final static int GO_NEXT = 0xAA;

    @Override
    public int getLayoutId(Bundle savedInstanceState) {
        return R.layout.activity_welcome;
    }

    @Override
    public void initEnv() {
        showHead(false, false);
        Observable.just(GO_NEXT)
                .delay(3000, TimeUnit.MILLISECONDS)
                .subscribe(new Consumer<Integer>() {
                    @Override
                    public void accept(Integer integer) throws Exception {
                        if (integer == GO_NEXT) {
                            toPage(KpActivity.class);
                        }
                    }
                });
    }
}

File: app/src/main/java/com/zhangw/aliencat/ui/welcome/WelcomeActivity.kt
package com.zhangw.aliencat.ui.welcome

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.zhangw.aliencat.R
import kotlinx.android.synthetic.main.activity_welcome.*

class WelcomeActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_welcome)

        // Example of a call to a native method
        sample_text.text = "alienCat"

        sample_text.setOnClickListener(View.OnClickListener {
            startActivity(Intent(this@WelcomeActivity, KpActivity::class.java))
        })
    }
}
